test/test_pebble.py

```diff
@@ -3185,6 +3185,82 @@ def test_get_notices_filters(self, client: MockClient):
             ('GET', '/v1/notices', query, None),
         ]
 
+    def test_get_identities(self, client: MockClient):
+        client.responses.append({
+            'result': {
+                'alice': {'access': 'admin', 'local': {'user-id': 42}},
+                'web': {'access': 'metrics', 'basic': {'password': 'hashed password'}},
+            },
+            'status': 'OK',
+            'status-code': 200,
+            'type': 'sync',
+        })
+
+        identities = client.get_identities()
+        assert identities == {
```

Comment:
I think we should use the dataclass here, rather than `from_dict`, as we don't need to be testing that as well here.

I'd also suggest we add a `str` test to test the `(str, Enum)`-style enum:

```
assert identities['alice']['access'] == 'admin'
assert identities['web']['access'] == 'metrics'
```